oaknational/Oak-Web-Application
# Feat/lesq 1040/optional onboarding step [LESQ-1040]

## Description

Music year: {owa_music_year}

- Adds a new page to to the onboarding journey 'How can oak support you'
- Optional list of multi select checkboxes
- Adds Skip button and subheading to onboarding form
- Only navigate to new page when not already signed up in hubspot

As this page depends on query params in the URL to get the data to send to hubspot, if a user were to navigate to this page directly or manually remove the query params form the URL (for whatever reason) the form will become invalid. In this case I've added an error message and a link to go back to the previous step (see screenshot below). @Monika-Koziol and @jack-skerrett-oak would be good to get your thoughts on this 🙏 

## How to test

1. Go to https://deploy-preview-2764--oak-web-application.netlify.thenational.academy/onboarding *while signed in*
2. Click on Yes
3. Fill out the school picker or manually enter school and continue

**Scenario 1**: Not subscribed to communications in Hubspot
4. You should be directed to the 'how can oak support you page'
5. Selecting options and clicking continue will send your selected options to hubspot
6. You should be able to click 'skip' which will submit your existing answers to hubspot

**Scenario 2:** Already subscribed to communications in Hubspot (no checkbox visible on the page)
4. You should be directed back to the home page without seeing the How can oak support you page
5. You hubspot submision will not include any data on how can oak support you

**Scenario 3**: Navigate directly to /onboarding/how-can-oak-support-you or clear some/all of the query params from the URL
4. You should see an error message
5. You should not be able to continue or skip (buttons disabled)
6. You should be able to click 'go back' in the error message or go back in the browser and start again from the school select page

## Screenshots

How it should now look:
<img width="600" alt="Screenshot 2024-09-10 at 10 21 17" src="https://github.com/user-attachments/assets/33047503-1755-4043-9b06-0f628551bab1">

Error state:
<img width="600" alt="Screenshot 2024-09-10 at 10 42 23" src="https://github.com/user-attachments/assets/baafb049-d888-4c3c-8263-ef8c29abb2e7">
